core/drivers/crypto/hisilicon/hpre_dh.c

```diff
@@ -0,0 +1,375 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2022, HiSilicon Technologies Co., Ltd.
+ * Kunpeng hardware accelerator hpre dh algorithm implementation.
+ */
+
+
+#include <hpre_dh.h>
+#include <hpre_main.h>
+#include <rng_support.h>
```

Comment:
```patch
+#include <crypto/crypto.h>`
 #include <hpre_dh.h>
 #include <hpre_main.h>
+#include <initcall.h>
+#include <malloc.h>
 #include <rng_support.h>
+#include <string.h>
+#include <trace.h>
+#include <util.h>

Reply:
I think these head files has been already included, such as string.h included in hisi_qm.h